api_docs/group-setting-values.md

```diff
@@ -4,6 +4,13 @@ Settings defining permissions in Zulip are increasingly represented
 using [user groups](/help/user-groups), which offer much more flexible
 configuration than the older [roles](/api/roles-and-permissions) system.
 
+!!! warn ""
+
+    The settings can only be set to a named user group currently.
+    So, only integer user group ID format and an object with empty
+    `direct_member_ids` and with `direct_subgroup_ids` containing a
+    single user group ID are supported.
```

Comment:
Not sure whether we want to mention about the server level setting here or just explain the behavior.

Reply:
Made the following tweaks -- I think conceptually this has been true since we added these in the first place, and we should just plan to have a `Changes` entry for when we we turn it on fully.

``` diff
diff --git a/api_docs/group-setting-values.md b/api_docs/group-setting-values.md
index 9e3a5d3ccb..62a20b8aac 100644
--- a/api_docs/group-setting-values.md
+++ b/api_docs/group-setting-values.md
@@ -6,10 +6,10 @@ configuration than the older [roles](/api/roles-and-permissions) system.
 
 !!! warn ""
 
-    The settings can only be set to a named user group currently.
-    So, only integer user group ID format and an object with empty
-    `direct_member_ids` and with `direct_subgroup_ids` containing a
-    single user group ID are supported.
+    This API feature is under development, and currently only values that
+    correspond to a single named user group are permitted in
+    production environments, pending the web application UI supporting
+    displaying more complex values correctly.
 
 In the API, these settings are represented using a **group-setting
 value**, which can take two forms:
diff --git a/zproject/default_settings.py b/zproject/default_settings.py
index 47bc2bd0b5..acadf16a5b 100644
--- a/zproject/default_settings.py
+++ b/zproject/default_settings.py
@@ -636,8 +636,8 @@ SIGNED_ACCESS_TOKEN_VALIDITY_IN_SECONDS = 60
 
 CUSTOM_AUTHENTICATION_WRAPPER_FUNCTION: Optional[Callable[..., Any]] = None
 
-# Whether we allow settings to be set to a collection of users
-# and groups. Set to False currently as we can only handle named
-# user groups in the UI and will be removed once we have that.
-# We allow the collection of users and groups only in tests for now.
+# Whether we allow settings to be set to a collection of users and
+# groups as described in api_docs/group-setting-values.md. Set to
+# False in production, as we can only handle named user groups in the
+# web app settings UI.
 ALLOW_ANONYMOUS_GROUP_VALUED_SETTINGS = False
diff --git a/zproject/dev_settings.py b/zproject/dev_settings.py
index 5c132426ca..c35015b60b 100644
--- a/zproject/dev_settings.py
+++ b/zproject/dev_settings.py
@@ -207,3 +207,6 @@ SCIM_CONFIG: Dict[str, SCIMConfigDict] = {
 SELF_HOSTING_MANAGEMENT_SUBDOMAIN = "selfhosting"
 DEVELOPMENT_DISABLE_PUSH_BOUNCER_DOMAIN_CHECK = True
 PUSH_NOTIFICATION_BOUNCER_URL = f"http://push.{EXTERNAL_HOST}"
+
+# Breaks the UI if used, but enabled for development environment testing.
+ALLOW_ANONYMOUS_GROUP_VALUED_SETTINGS = True
```